### PR TITLE
'Esc' works for a vertical volume bar and menus

### PR DESCRIPTION
**Pull request by gjanblaszczyk**
## Description
Following #6004 , this PR adds support for 'Esc' functionality in the volume bar and menus.

## Specific Changes proposed
The PR improves player accessibility by adding 'Esc' functionality to the volume panel and menu popups.  

## Requirements Checklist
- [X] Feature implemented / Bug fixed
- [X] If necessary, more likely in a feature request than a bug fix
  - [X] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [ ] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/js/menu/menu-button.js [hunk 1]
```diff
@@ -49,9 +52,11 @@ class MenuButton extends Component {
     this.on(this.menuButton_, 'click', this.handleClick);
     this.on(this.menuButton_, 'keydown', this.handleKeyDown);
     this.on(this.menuButton_, 'mouseenter', () => {
+      this.addClass('vjs-hover');
       this.menu.show();
+      Events.on(document, 'keyup', Fn.bind(this, this.handleMenuKeyUp));
     });
-
+    this.on('mouseleave', this.handleMouseLeave);
```
[the remark this answers is not in the document]
Comment (gjanblaszczyk) [reply src/js/menu/menu-button.js]: The `mouseleave` event should be attached to `this` element. If we attach it to the `menuButton` then we are not able to move mouse pointer between the menu button and menu popup without hiding the menu popup itself.
hover

File: src/js/menu/menu-button.js [hunk 2]
```diff
@@ -49,9 +52,11 @@ class MenuButton extends Component {
     this.on(this.menuButton_, 'click', this.handleClick);
     this.on(this.menuButton_, 'keydown', this.handleKeyDown);
     this.on(this.menuButton_, 'mouseenter', () => {
+      this.addClass('vjs-hover');
       this.menu.show();
+      Events.on(document, 'keyup', Fn.bind(this, this.handleMenuKeyUp));
```
[the remark this answers is not in the document]
Comment (gjanblaszczyk) [reply src/js/menu/menu-button.js]: Good catch, I will clean up this handler.

File: src/js/control-bar/volume-panel.js [hunk 3]
```diff
@@ -109,6 +118,65 @@ class VolumePanel extends Component {
...
+   * calls `this.handleMouseOut`.
+   *
+   * @param {EventTarget~Event} event
+   *        The `mouseout` event that caused this function to be called.
+   *
+   * @listens mouseout
+   */
+  handleMouseOut(event) {
+    this.removeClass('vjs-hover');
+    Events.off(document, 'keyup', Fn.bind(this, this.handleKeyPress));
```
[the remark this answers is not in the document]
Comment (gjanblaszczyk) [reply src/js/control-bar/volume-panel.js]: Ok, I will clean up this handler.

**Comment by gjanblaszczyk** [comment]
Hi @gkatsev, Yes, exactly. I had the same issue that was recently reported by `marcelo` on Slack. So I decided to fix it almost 2 months ago and this PR contains a fix for it.
btw Thanks for the review.

File: src/js/control-bar/volume-panel.js [hunk 4]
```diff
@@ -109,6 +118,73 @@ class VolumePanel extends Component {
...
+  }
+
+  /**
+   * Handles `keydown|keyup` events on the document, looking for ESC, which closes
+   * the volume panel.
```
Comment (misteroneill) [inline src/js/control-bar/volume-panel.js]: Looks like a copy/paste issue here.

File: src/js/control-bar/volume-panel.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (misteroneill) [reply src/js/control-bar/volume-panel.js]: I wonder if these two lines should be moved to a `cleanup` method, which can be called whenever this change is needed.

File: src/js/control-bar/volume-panel.js [hunk 5]
```diff
@@ -109,6 +118,73 @@ class VolumePanel extends Component {
...
+   *
+   * @param {EventTarget~Event} event
+   *        The keypress that triggered this event.
+   *
+   * @listens keydown | keyup
+   */
+  handleKeyPress(event) {
+    if (keycode.isEventKey(event, 'Esc')) {
+      this.removeClass('vjs-hover');
```
Comment (misteroneill) [inline src/js/control-bar/volume-panel.js]: I think we also need to remove the `document`-level listener here?

File: src/js/control-bar/volume-panel.js (hunk 4 above)
Comment (gjanblaszczyk) [reply src/js/control-bar/volume-panel.js]: Thanks, nice catch! I have just fixed the comment.

File: src/js/control-bar/volume-panel.js (hunk 5 above)
Comment (gjanblaszczyk) [reply src/js/control-bar/volume-panel.js]: Thanks, I have just improved the method following your comment.

**Pull request merged**